sound/soc/sof/ipc4.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ *
+ * Author: Rander Wang <rander.wang@linux.intel.com>
+ */
+
+#ifndef __SOF_IPC4_H
+#define __SOF_IPC4_H
+
+struct sof_ipc4_fw_status {
+	int status;
+	char *msg;
+};
+
+static inline int sof_ipc4_module_buffer_size(int ch,  int rate, int width, int sch_num)
+{
+	/* 1ms data size in fw */
+	return (ch) * ((rate) / 1000) * ((width) >> 3) * sch_num;
```

Comment:
that's broken for 44.1 !

Reply:
Indeed, it is not accurate. I don't know how should we handle this. Say if e.g. ch=2, rate=44100, width=16, sch_num=2. The accurate value is 352.8, but buffer size is an integer and will be 352. @RanderWang What do you think?

Reply:
yes, we need to allocated buffer with rate of 48k for 44.1k. Now we don't support 44.1 in FW and we need to convert it to 48k. And even we can support native 44.1, a little bigger buffer is no harmful.